### DISCUSSION TITLE
can you build the [libarchive] wasm via browser

### DISCUSSION THREAD

**Discussion by nenge123** [gildas-lormeau/zip.js#383] [General]
https://github.com/libarchive/libarchive

that support rar/7z

**Comment by gildas-lormeau** [comment]
Probably, I never built JS code with emscripten. I guess the resulting file would be very large though.

**Comment by nenge123** [comment]
yes, but i can make a zip file. using zip.js to load it. the size will less than 300KB

-----------
that oldest repo,but it only support ascii chart code, like the chinese name all be change to "*"

https://github.com/nika-begiashvili/libarchivejs

**Reply by gildas-lormeau** [reply]
It looks like it's a bug in libarchive, see https://github.com/libarchive/libarchive/issues/587

**Comment by nenge123** [comment]
https://www.rarlab.com/rar/unrarsrc-6.2.5.tar.gz

can you make this one to a wasm?

the last one (https://github.com/wcchoi/libunrar-js) support RAR5,but too big.

other one (https://github.com/tnikolai2/libunrar-js/)  , filesize less, but have a bug ,if filename char byte>127 will return empty filename.
